src/java.base/share/classes/java/lang/invoke/InnerClassLambdaMetafactory.java

```diff
@@ -204,12 +216,102 @@ public InnerClassLambdaMetafactory(MethodHandles.Lookup caller,
     }
 
     private static String lambdaClassName(Class<?> targetClass) {
+        return createNameFromTargetClass(targetClass) + counter.incrementAndGet();
+    }
+
+    private static String createNameFromTargetClass(Class<?> targetClass) {
         String name = targetClass.getName();
         if (targetClass.isHidden()) {
             // use the original class name
             name = name.replace('/', '_');
         }
-        return name.replace('.', '/') + "$$Lambda$" + counter.incrementAndGet();
+        return name.replace('.', '/') + "$$Lambda$";
+    }
+
+    /**
+     * Calculate hash value of the given String in the same manner as the
+     * java.lang.StringUTF16#hashCode does with difference that hash value
+     * is long instead of int.
+     *
+     * @param name String for which method calculates hash value for
+     *
+     * @return a hash value for the given String
+     * */
+    private String fixedSizeStringHash(String name) {
+        long h = 0;
+        int length = name.length();
+        for (int i = 0; i < length; i++) {
+            h = 31 * h + name.charAt(i);
+        }
+
+        StringBuilder hash = new StringBuilder(Long.toString(Math.abs(h), Character.MAX_RADIX));
```

Comment:
Define a private static method
```java
String hashValueString(long hashValue) {
    return Long.toString(hashValue, Character.MAX_RADIX);
}
```
and use the method in the static initializer block and also here.

Reply:
Method defined and used in proper places.